.azure-pipelines/common.yml

```diff
@@ -0,0 +1,115 @@
+parameters:
+  name: ''
+  platform: ''
+  settings: ''
+  py_3_5: false
+  py_3_6: false
+  py_3_7: true
+  py_3_8: false
+  container_name: ''
+  container_startup: ''
+  additional_apt_gets: ''
+  additional_requirements: ''
+
+jobs:
+- job: ${{ format('{0}_{1}', parameters.name, parameters.platform) }}
+
+  timeoutInMinutes: 360
+
+  pool:
+    ${{ if eq(parameters.platform, 'Windows') }}:
+      vmImage: VS2017-Win2016
+    ${{ if eq(parameters.platform, 'Linux') }}:
+      vmImage: Ubuntu 16.04
+
+  strategy:
+    matrix:
+      ${{ if eq(parameters.py_3_5, 'true') }}:
+        Python35:
+          PYTHON_VERSION: '3.5'
+      ${{ if eq(parameters.py_3_6, 'true') }}:
+        Python36:
+          PYTHON_VERSION: '3.6'
+      ${{ if eq(parameters.py_3_7, 'true') }}:
+        Python37:
+          PYTHON_VERSION: '3.7'
+      ${{ if eq(parameters.py_3_8, 'true') }}:
+        Python38:
+          PYTHON_VERSION: '>= 3.8.0a'
+
+  steps:
+    - ${{ if ne(parameters.container_startup, '') }}:
+      - bash: ${{ parameters.container_startup }}
+        displayName: Spin up container
+
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(PYTHON_VERSION)'
+        architecture: 'x64'
+      displayName: Use Python version $(PYTHON_VERSION)
+
+    - ${{ if eq(parameters.platform, 'Linux') }}:
+      - bash: |
+          sudo apt-get -yq update
+          sudo apt-get -yq install libmemcached-dev
+          if [ -n "$OTHER_APT" ]; then
+            sudo apt-get -yq install $OTHER_APT
+          fi
+        displayName: Setup Linux environment
+        env:
+          OTHER_APT: ${{ parameters.additional_apt_gets }}
+
+    - bash: |
+        python -m pip install --upgrade pip setuptools wheel
+        pip install unittest-xml-reporting

```

Comment:
Same as above

Reply:
Same - will look into having the `unittest-xml-reporting` package ready to go. The upgrades to pip, setuptools, and wheel were kind of a defensive move against images sliding out of date, but they aren't strictly necessary.